pkg/vpcmodel/nodesExplainability.go

```diff
@@ -56,66 +158,141 @@ func (c *VPCConfig) getFiltersEnablingRulesBetweenNodesPerDirectionAndLayer(
 	return &rulesOfFilter, nil
 }
 
-func (c *VPCConfig) GetRulesOfConnection(src, dst Node) (rulesOfConnection *RulesOfConnection, err error) {
+func (c *VPCConfig) getRulesOfConnection(src, dst Node) (rulesOfConnection *rulesConnection, err error) {
 	filterLayers := []string{SecurityGroupLayer}
-	rulesOfConnection = &RulesOfConnection{make(rulesInLayers, len(filterLayers)),
+	rulesOfConnection = &rulesConnection{make(rulesInLayers, len(filterLayers)),
```

Comment:
can initialize `rulesOfConnection = &rulesConnection{}`  , since you later assign the fields `ingressRules`, `egressRules` from other objects.

Reply:
done